### PR TITLE
Test::Quattor set_service_variant support reload and additional actions

### PR DESCRIPTION
**Pull request by stdweird**
Requires https://github.com/quattor/CAF/pull/118

### CODE REVIEW

**Comment by ned21** [comment]
LGTM

**Comment by stdweird** [comment]
@ned21 this needs rereviewing and https://github.com/quattor/CAF/pull/118

File: build-scripts/src/main/perl/Test/Quattor.pm [hunk 1]
```diff
@@ -173,11 +171,11 @@ my @command_history = ();
 
 =item * C<caf_file_close_diff>
 
-A boolean to mimick the regular (i.e. when no C<NoAction> is set) behaviour of a 
-C<CAF::FileWriter> or C<CAF::FileEditor> C<close> (it returns wheter or not the 
+A boolean to mimick the regular (i.e. when no C<NoAction> is set) behaviour of a
```
Comment (ned21) [inline build-scripts/src/main/perl/Test/Quattor.pm]: s/mimick/mimic/, wheter => whether at line 175, chanegs => changes in line 178

File: build-scripts/src/main/perl/Test/Quattor.pm [hunk 2]
```diff
@@ -584,9 +612,14 @@ sub set_service_variant
             croak "Unsupported variant $variant";
         }
     }
+
+    # not for the faint of heart
```
Comment (ned21) [inline build-scripts/src/main/perl/Test/Quattor.pm]: This comment doesn't say much  - perhaps provide a bit of explanation?

Comment (stdweird) [inline build-scripts/src/main/perl/Test/Quattor.pm]: i'm blindly assuming the namespace is usable. 
i would actually rather get rid of this whole force_service_variant. (it completly bypasses AUTOLOAD, but i think this is becasue old code didn't have os_flavours as separate method.) but it was what the old code did, so i'm keeping it around a bit.

Comment (stdweird) [inline build-scripts/src/main/perl/Test/Quattor.pm]: (suggestions welcome for rewording :smile: )

**Comment by jrha** [comment]
@stdweird test failing :crying_cat_face:

**Comment by stdweird** [comment]
@jrha also the ones from ncm-postgres fail, i have to investigate

**Comment by stdweird** [comment]
@jrha i might have forgotten to push the last commit . i also need new build tools with this PR in to pass the postgres tests

File: build-scripts/src/main/perl/Test/Quattor.pm (hunk 2 above)
Comment (jrha) [inline build-scripts/src/main/perl/Test/Quattor.pm]: How about `# Blindly assume that the namespace is usable.`

**Comment by stdweird** [comment]
@jrha comment reworded per your suggestion, thx

**Pull request merged**